hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,75 @@
+defmodule State do
+    defstruct game_state: :initializing,
+     turns_left: 7,
+     letters: [],
+     used: [],
+     word: "",
+     last_guess: ""
+end
+
+defmodule Hangman.Game do
+  @moduledoc """
+  The Hangman.Game module provides an api to allow users to create a game of
+  hangman, make guesses, and see the progression off the game.
+  """
+
+  def new_game() do
+    new_game(Dictionary.random_word())
+  end
+
+  # Exists for testing purposes
+  def new_game(word) do
+    %State{word: word, letters: map_to_letters(word)}
+  end
+
+  def tally(game), do: Map.delete(game, :word)
```

Comment:
## -1

So, this is picky, but simply returning the internal state, minus the word, is a little dangerous. It works for now, but in the real world someone coming along later could add a field to the state and not realize that it would by default be copied out through the API.

Reply:
Not even picky - that's actually a really good point that i've never thought of.  I will keep this in mind in my future design.